app/src/main/java/free/rm/skytube/businessobjects/Sponsorblock/SBTasks.java

```diff
@@ -0,0 +1,98 @@
+package free.rm.skytube.businessobjects.Sponsorblock;
+
+import android.content.Context;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import free.rm.skytube.BuildConfig;
+import free.rm.skytube.R;
+import free.rm.skytube.app.SkyTubeApp;
+import free.rm.skytube.businessobjects.YouTube.newpipe.VideoId;
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.core.Maybe;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+public class SBTasks {
+    private static final String TAG = SBTasks.class.getSimpleName();
+
+    /**
+     * A task that retrieves information from the Sponsorblock API about a youtube video
+     *
+     * @param videoId The ID of the youtube video to watch
+     */
+    public static Maybe<SBVideoInfo> retrieveSponsorblockSegments(@NonNull Context context, @NonNull VideoId videoId) {
+        return Maybe.fromCallable(() -> {
+            String[] defaultFilterList = SkyTubeApp.getStringArray(R.array.sponsorblock_filtering_list_default_values);
+            Set<String> filterList = SkyTubeApp.getPreferenceManager().getStringSet("pref_key_sponsorblock_category_list", new HashSet<String>(Arrays.asList(defaultFilterList)));
```

Comment:
Could you please move this settings related code to the Settings class? 
Unfortunately not everything located there, but I generally add new things there,